### PR TITLE
Follow up in PR #46

### PR DESCRIPTION
**Pull request by ctron**
As I wasn't able to push to the other PR, here is a follow up.

It basically the same, just adding the second test, verifying the PR. When doing that I encountered the issue that SHA1 isn't supported anymore. So I introduced a parameter to allow providing the hash algorithm. Which makes sense anyway.

### CODE REVIEW

**Comment by ctron** [comment]
@mat1e Maybe you can have a look too. These are still your commits, rebased to the most recent main branch and adding one commit at the end.

**Comment by ctron** [comment]
@dwalluck My plan is to merge the other PR first, then this one. I might do that today, as I will be off the next week, and I don't want this lying around for much longer.

**Comment by mat1e** [comment]
We would have been disturbed by the non configurable hash, I had not identified this... Thanks you @ctron to continu the work. 
I saw something but I don't know if it will fail the test.

**Comment by ctron** [comment]
> I saw something but I don't know if it will fail the test.

The tests seem to pass in the CI, which did you notice?

I anyone ok merging it?

File: rpm/src/test/java/org/eclipse/packager/rpm/signature/RpmFileSignatureProcessorTest.java [hunk 1]
```diff
@@ -7,45 +7,55 @@
...
 
     private static final String SOURCE_FILE_PATH = "src/test/resources/data/org.eclipse.scada-0.2.1-1.noarch.rpm";
+
     private static final String PRIVATE_KEY_PATH = "src/test/resources/key/private_key.txt";
+
     private static final String PUBLIC_KEY_PATH = "src/test/resources/key/public_key.txt";
-    private static final String RESULT_FILE_PATH = "src/test/resources/result/org.eclipse.scada-0.2.1-1.noarch.rpm";
-    private static final String RESULT_DIR = "src/test/resources/result";
+
+    private static final String RESULT_DIR = "target/test-data/signature";
```
Comment (dwalluck) [inline rpm/src/test/java/org/eclipse/packager/rpm/signature/RpmFileSignatureProcessorTest.java:52]: @ctron OK, some tests are writing directly to target/ which isn't really good and could possibly cause issues with repeated runs. I will create a PR to use JUnit 5 `@TempDir` for the output directory for any tests.

**Pull request merged**